ml/rl/workflow/generate_batch.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+import logging
+from ml.rl.workflow.model_managers.union import ModelManager__Union
+from ml.rl.training.dqn_trainer import DQNTrainer
+from ml.rl.gym.policies.policy import RandomPolicy
+import pdb
+from ml.rl.training.rl_dataset import RLDataset
+from ml.rl.gym.preprocessors.action_preprocessors.action_preprocessor import (
+    continuous_action_preprocessor,
+    discrete_action_preprocessor,
+)
+from ml.rl.gym.agents.post_step import log_data_post_step
+from ml.rl.workflow.types import RewardOptions
+from ml.rl.gym.tests.test_gym import build_normalizer
+from ml.rl.gym.runners.gymrunner import run_episode
+import gym
+import pandas as pd
+from ml.rl.workflow.types import TableSpec
+import pyspark
+
+
+logger = logging.getLogger(__name__)
+
+
+def offline_gym(
+    env: str,
+    pkl_path: str,
+    model: ModelManager__Union,
+    num_episodes: int,
+    max_steps: int,
+):
+    env = gym.make(env)
+    normalization = build_normalizer(env)
+    logger.info(f"Normalization is {normalization}")
+
+    manager = model.value
+    actions = manager.trainer_param.actions
+    logger.info(f"Actions are {actions}")
+
+    manager.initialize_trainer(
+        use_gpu=False,
+        reward_options=RewardOptions(),
+        normalization_data_map=normalization,
+    )
+    trainer = manager.trainer
+    policy = RandomPolicy(len(actions))
+
+    # case on the different trainers
+    if isinstance(trainer, DQNTrainer):
+        action_preprocessor = discrete_action_preprocessor
+    else:
+        logger.info(f"trainer has type {type(trainer)} not supported!")
+        import pdb
+
+        pdb.set_trace()
+
+    dataset = RLDataset(pkl_path)
+    reward_history = []
+    for i in range(num_episodes):
+        logger.info(f"running episode {i}")
+        post_step = log_data_post_step(dataset, str(i))
+        ep_reward = run_episode(
+            env=env,
+            policy=policy,
+            action_preprocessor=action_preprocessor,
+            post_step=post_step,
+            max_steps=max_steps,
+        )
+        reward_history.append(ep_reward)
+
+    logger.info(f"Saving dataset with {len(dataset)} samples to {pkl_path}")
+    df = dataset.to_pandas_df()
+    pd.to_pickle(df, pkl_path)
+
+
+# TODO: remove after refactoring and having spark_utils
+def get_spark_session(master: str = "local[1]"):
+    """ Get a spark session """
+    spark = (
+        pyspark.sql.SparkSession.builder.master(master)
+        .enableHiveSupport()
+        .getOrCreate()
+    )
+    return spark
+
+
+def upload_to_hive(pkl_path: str, table_spec: TableSpec):
```

Comment:
Can we make RLDataset save parquet & load parquet file here?

Reply:
pyarrow cannot handle maps, sticking with pickle as discussed